modules/cli/src/main/scala/scala/cli/commands/ScalaCommand.scala

```diff
@@ -202,4 +202,56 @@ abstract class ScalaCommand[T](implicit myParser: Parser[T], help: Help[T])
           // This requires writing our own minimal JNI library, that publishes '.a' files too for static linking in the executable of Scala CLI.
           None
       }
+
+  /** @param options
+    *   command-specific [[T]] options
+    * @return
+    *   Tries to create BuildOptions based on [[sharedOptions]] and exits on error. Override to
+    *   change this behaviour.
+    */
+  def buildOptions(options: T): Option[BuildOptions] =
+    sharedOptions(options).map(shared => shared.buildOptions().orExit(shared.logger))
+
+  protected def buildOptionsOrExit(options: T): BuildOptions =
+    buildOptions(options) match {
+      case Some(bo) => bo
+      case _ =>
+        sharedOptions(options).foreach(_.logger.debug("build options could not be initialized"))
+        sys.exit(1)
+    }
+
+  /** @param options
+    *   command-specific [[T]] options
+    * @return
+    *   by default [[sharedOptions]].logging, override to adjust.
+    */
+  def loggingOptions(options: T): Option[LoggingOptions] =
```

Comment:
I think `loggingOptions` should always be set, why does this function return `Option`?
```suggestion
  def loggingOptions(options: T): LoggingOptions =
```
This should force everyone to implement `loggingOptions ` for new commands.

Reply:
I considered if it's by design, actually, but we do have sub-commands which do not support `LoggingOptions`. An example is the `about` sub-command.
If we made all those have `LoggingOptions`, then this wouldn't have to be an option, strictly.
then a definition for easy access to the logger could be added as well.

However, the benefit of leaving it as an option is that it can depend on `sharedOptions` by default.

Reply:
However, if we decide to enforce all options to have `loggingOptions`, I'd rather do that in a separate PR